File: app/java/src/main/java/com/example/production/domain/model/subcontract/SupplyDetail.java
package com.example.production.domain.model.subcontract;

import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 支給明細データ
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class SupplyDetail {
    private Integer id;
    private String supplyNumber;
    private Integer lineNumber;
    private String itemCode;
    private BigDecimal quantity;
    private BigDecimal unitPrice;
    private BigDecimal amount;
    private String remarks;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: app/csharp/src/ProductionManagement.Domain/Models/Subcontract/SupplyDetail.cs
namespace ProductionManagement.Domain.Models.Subcontract;

/// <summary>
/// 支給明細データ
/// </summary>
public class SupplyDetail
{
    public int Id { get; set; }
    public required string SupplyNumber { get; set; }
    public required int LineNumber { get; set; }
    public required string ItemCode { get; set; }
    public required decimal Quantity { get; set; }
    public required decimal UnitPrice { get; set; }
    public required decimal Amount { get; set; }
    public string? Remarks { get; set; }
    public DateTime? CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
}
